src/librustc_trans/back/write.rs

```diff
@@ -807,7 +808,7 @@ fn binaryen_assemble(cgcx: &CodegenContext,
     if cgcx.debuginfo != config::NoDebugInfo {
         options.debuginfo(true);
     }
-    if cgcx.crate_types.contains(&config::CrateTypeExecutable) {
+    if cgcx.wasm_auto_run && cgcx.crate_types.contains(&config::CrateTypeExecutable) {
```

Comment:
Correct me if I'm wrong, but doesn't this change the current behaivor so that **without** specifying the `#![wasm_auto_run]` attribute the `start` section won't be emitted?

Reply:
Yes, I mentioned this in the top post.

As I'm sure you're aware, there's no way to call into wasm or read memory until the main function has exited. (Well, you *can* read memory, but only if the wasm file is compiled to import it).

Anyone using the wasm target is likely to run head first into this problem as reading memory is necessary for any kind of sane interop between the host and wasm, so it made more sense to me to have the default be to *not* automatically run the main function, and instead explicitly call it via `module.exports.main()`. The old behaviour can still be obtained via the `wasm_auto_run` attribute.

Reply:
Ah right, you've edited the explanation for this in the original post. Sorry for not noticing!

I just wanted to make sure that this means what I though it means. Yes, I also think the new behavior is a lot more sane; I'm not sure what was the original idea behind the `start` entrypoint in WASM coupled with how Web browsers load WASM, but as it stands it's - just as you've said - almost completely useless since you can't do anything useful in it.